packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -2098,3 +2100,252 @@ describe("SharedString interval collections", () => {
 		});
 	});
 });
+
+describe("the start and end positions of intervals are updated in response to edits", () => {
+	const stringFactory = new SharedStringFactory();
+	let containerRuntimeFactory: MockContainerRuntimeFactoryForReconnection;
+	let clients: [Client, Client, Client];
+	beforeEach(() => {
+		containerRuntimeFactory = new MockContainerRuntimeFactoryForReconnection();
+		clients = constructClients(
+			containerRuntimeFactory,
+			2,
+			{
+				intervalStickinessEnabled: true,
+				mergeTreeEnableObliterate: true,
+				mergeTreeReferencesCanSlideToEndpoint: true,
+			},
+			stringFactory,
+		);
+	});
+
+	const removeRanges: {
+		interval: [SequencePlace, SequencePlace];
+		removeRange: [number, number];
+		expected: [SequencePlace, SequencePlace];
+		skip?: string[];
+	}[] = [
+		// remove the entire interval
+		// end should slide to beginning, resulting in an empty range
+		{
+			interval: ["start", 1],
+			removeRange: [0, 1],
+			expected: ["start", 0],
+			// TODO: #8111: enable after interval side is correctly loaded from summary for endpoints at start or end.
+			skip: ["slide interval loaded from summary"],
+		},
+		{
+			interval: [2, 5],
+			removeRange: [2, 5],
+			expected: [2, 2],
+		},
+		{
+			interval: [
+				{ pos: 5, side: Side.After },
+				{ pos: 9, side: Side.Before },
+			],
+			removeRange: [6, 9],
+			expected: [
+				{ pos: 5, side: Side.After },
+				{ pos: 6, side: Side.Before },
+			],
+		},
+		{
+			interval: [8, "end"],
+			removeRange: [8, 10],
+			expected: ["end", "end"],
+			skip: [
+				// TODO: #8112: re-enable once empty intervals are no longer skipped when using the new slide behavior.
```

Comment:
Note for reviewers:
See https://github.com/microsoft/FluidFramework/pull/21230 for a possible resolution.
This may also be considered an issue with the test if the results of `getIntervalCollectionLabels` are not guaranteed to be eventually consistent between clients.